pkg/duck/listable.go

```diff
@@ -156,6 +162,45 @@ func (t *listableTracker) TrackInNamespaceKReference(ctx context.Context, obj me
 	}
 }
 
+// TrackInNamespace satisfies the ListableTracker interface.
+func (t *listableTracker) Track(ctx context.Context, obj metav1.Object, namespace string) Track {
```

Comment:
A question: why having the new Track function here?

Reply:
This package provides a set of tools for tracking objects/references. Since we can no longer rely on the assumption that these will always be in the same namespace, it makes sense to update the interface to have methods for both of the existing types of tracking we do. I'd rather add it now while I remember than have someone else need to figure this out again in the future